doc/designs/handling-some-max-defines.md

```diff
@@ -0,0 +1,149 @@
+Handling Some MAX Defines in Future
+===================================
+
+Problem Definition
+------------------
+
+The public headers contain multiple `#define` macros that limit sizes or
+numbers of various kinds. In some cases they are uncontroversial so they
+do not require any changes or workarounds for these limits. Such values
+are not discussed further in this document. This document discusses only
+some particularly problematic values and proposes some ways how to
+change or overcome these particular limits.
+
+Individual Values
+-----------------
+
+### HMAC_MAX_MD_CBLOCK
+
+**Current value:** 200
+
+This is a deprecated define which is useless. It is not used anywhere.
+
+#### Proposed solution:
+
+It should be just removed with 4.0.
+
+### EVP_MAX_MD_SIZE
+
+**Current value:** 64
+
+It is unlikely we will see longer than 512 bit hashes any time soon.
+XOF functions do not count and the XOF output length is not and should
+not be limited by this value.
+
+It is widely used throughout the codebase and by 3rd party applications.
+
```

Comment:
Whats the proposed solution here?  convert users of this macro to EVP_MD_get_size and remove the macro for 4.0?